includes/admin/class-wc-admin-notices.php

```diff
@@ -64,6 +64,52 @@ public static function init() {
 		}
 	}
 
+	/**
+	 * Parses query to create nonces when available.
+	 *
+	 * @param object $response The WP_REST_Response we're working with.
+	 * @return object $response The prepared WP_REST_Response object.
+	 */
+	public static function prepare_note_with_nonce( $response ) {
+		if ( 'wc-update-db-reminder' !== $response->data['name'] || ! isset( $response->data['actions'] ) ) {
+			return $response;
+		}
+
+		foreach ( $response->data['actions'] as $action_key => $action ) {
+			$url_parts = ! empty( $action->query ) ? wp_parse_url( $action->query ) : '';
+
+			if ( ! isset( $url_parts['query'] ) ) {
+				$response->data['actions'][ $action_key ] = $action;
+				continue;
+			}
+
+			wp_parse_str( $url_parts['query'], $params );
+
+			if ( array_key_exists( '_nonce_action', $params ) && array_key_exists( '_nonce_name', $params ) ) {
+				$_params = $params;
+
+				unset( $_params['_nonce_action'] );
+				unset( $_params['_nonce_name'] );
+
+				$url = $url_parts['scheme'] . '://' . $url_parts['host'] . $url_parts['path'];
+
+				$parsed_query = wp_nonce_url(
+					add_query_arg(
+						$_params,
+						$url
+					),
+					$params['_nonce_action'],
+					$params['_nonce_name']
+				);
```

Comment:
Maybe we can just use use directly `add_query_arg()` instead of relying in `wp_nonce_url`, especially if later we end up calling `html_entity_decode`
```
$parsed_query = $parsed_query = add_query_arg(
					$params['_nonce_action'],
					wp_create_nonce( $params['_nonce_name'] ),
					add_query_arg(
						$_params,
						$url
					)
				);
```

To avoid potential security issues, we need to restrict what sort of values are allowed for `$params['_nonce_action']`, not sure if there's some sort of prefix we can enforce. Otherwise, this endpoint can be used to generate some other nonces not really related to this feature.

Reply:
Thanks for looking @xknown - so we do have a check here https://github.com/woocommerce/woocommerce/pull/28500/files#diff-16f554e0b9034833e30aa59fb984c8a67a1faadb5c9bc511db4e9e716c5556f5R74 that ensures we're taking action only on known items. But regardless I've pushed up changes to specifically whitelist the nonce action names.